src/gui/addnewtorrentdialog.cpp

```diff
@@ -429,6 +429,12 @@ AddNewTorrentDialog::AddNewTorrentDialog(const BitTorrent::TorrentDescriptor &to
     }
 
     TMMChanged(m_ui->comboTTM->currentIndex());
+
+#if defined(Q_OS_MACOS)
+    //By not setting a parent to the "addnewtorrentdialog",
+    //all those dialogs will be displayed on top and will not overlap with the main window.
+    setParent(nullptr);
```

Comment:
If solution with `parent` really works then it's better to apply it in the place where constructor is called instead of overwriting `parent` here.

Reply:
done.